### PR TITLE
Testing epic - (1) end-to-end tests for our API endpoints

### PR DESCRIPTION
**Pull request by thaigillespie**

### CODE REVIEW

**Comment by godwinpang** [comment]
closes #68

File: src/tests/event.controller.test.ts [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
-//       console.log(eventController.getMultipleEvents());
-//       expect(true).toEqual(true);
-//     });
-//   });
-// });
+import request from 'supertest';
+import { getExpressApp } from '../app';
+import { Connection } from 'typeorm';
+
+class currentApp {
```
[the remark this answers is not in the document]
Comment (thaigillespie) [reply src/tests/event.controller.test.ts:5]: Yeah I agree. I realized this when I was in bed trying to sleep, but it was really late already so I left it for today :))

**Comment by thaigillespie** [comment]
`import { App } from 'express';` gives an error for me. It says App doesn't exist there

**Comment by thaigillespie** [comment]
https://stackoverflow.com/questions/27676884/explicitly-specifying-types-for-express-application-request-response

**Comment by godwinpang** [comment]
this is really really cool - ik this is technically out of scope, but could we add a circleci workflow to actually run these tests?

**Comment by thaigillespie** [comment]
So just add in this PR as well? Sure I'll look into that

**Comment by godwinpang** [comment]
can we also add this into tsconfig so we don't compile the tests for prod? on my local machine it tries to run both the compiled js version and the ts tests

`"exclude": [
    "tests/*"
  ]`

**Comment by thaigillespie** [comment]
Sure

**Comment by thaigillespie** [comment]
@godwinpang the circleci test job needs our env vars from .env to run the tests. One way we can do this is to specify env vars in a job/run step (https://circleci.com/docs/2.0/env-vars/#environment-variable-usage-options). However, this means we would have to expose our env vars, which is not desirable. I'll see what I can do to not expose our env var values. Any ideas?

**Comment by thaigillespie** [comment]
Nvm found a way. I'll go with that for now

**Comment by thaigillespie** [comment]
I reverted the config.yml file back to the original for now. I'll work on this later, will work on FE migration now @godwinpang.

**Comment by godwinpang** [comment]
if you reverted circleci lets merge in 
@thaigillespie

**Comment by godwinpang** [comment]
wait can you squash first - ik merging in squashes it but uh it'd be nice not to have AAAAAAAA on master

**Pull request merged**